website/project/model.py

```diff
@@ -2974,13 +2986,20 @@ def to_json(self):
 
 
 class Sanction(StoredObject):
-    """Sanction object is a generic way to track approval states"""
-
+    """Sanction class is a generic way to track approval states"""
+    # Tell modularodm not to attach backends
     abstract = True
 
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    # Neither approved not cancelled
     UNAPPROVED = 'unapproved'
+    # Has approval
     APPROVED = 'approved'
+    # Rejected by at least one person
     REJECTED = 'rejected'
+    # One of 'unapproved', 'active', 'cancelled', or 'completed
+    state = fields.StringField(default=UNAPPROVED)
```

Comment:
Because this field is restricted to a very specific subset of items, I'd recommend making sure the value is in that list.

ModularODM doesn't seem to have a `choices` argument, but this can be done with a validator. See https://github.com/abought/osf.io/blob/develop/website/project/model.py#L610 for inspiration.